languagetool-core/src/main/java/org/languagetool/language/identifier/SimpleLanguageIdentifier.java

```diff
@@ -0,0 +1,147 @@
+
+    private final Map<String, SpellingCheckRule> spellingCheckRules = new HashMap<>();
+
+    public SimpleLanguageIdentifier() {
+        super(1000);
+        List<Language> languages = Languages.get();
+        for (Language language : languages) {
+            if (language.isVariant() || language.getShortCode().equals("zz")) {
+                continue;
+            }
+            boolean hasVariant = language.hasVariant();
+            Language spellingRuleLanguage = null;
+            if (hasVariant) {
+                spellingRuleLanguage = language.getDefaultLanguageVariant();
+            }
+            if (spellingRuleLanguage == null) {
+                spellingRuleLanguage = language;
+            }
+            SpellingCheckRule defaultSpellingCheckRule = spellingRuleLanguage.getDefaultSpellingRule(JLanguageTool.getDataBroker().getResourceBundle(JLanguageTool.MESSAGE_BUNDLE, new Locale(spellingRuleLanguage.getShortCodeWithCountryAndVariant())));
+            if (defaultSpellingCheckRule != null) {
+                spellingCheckRules.put(language.getShortCode(), defaultSpellingCheckRule);
+            } else {
+                log.warn("Could not found default speller rule for {}", language.getShortCode());
+            }
+        }
+    }
+
+    public SimpleLanguageIdentifier(List<String> preferredLangCodes) {
+        super(1000);
+        log.info("Init SpellcheckLangIdentifier with {}", preferredLangCodes);
+        preferredLangCodes.forEach(langCode -> {
+            Language language = Languages.getLanguageForShortCode(langCode);
+            SpellingCheckRule defaultSpellingRule = language.getDefaultSpellingRule(JLanguageTool.getDataBroker().getResourceBundle(JLanguageTool.MESSAGE_BUNDLE, new Locale(langCode)));
+            if (defaultSpellingRule != null) {
+                spellingCheckRules.put(language.getShortCode(), defaultSpellingRule);
+            }
+        });
+    }
+
+    @Nullable
+    @Override
+    public DetectedLanguage detectLanguage(String cleanText, List<String> noopLangsTmp, List<String> preferredLangsTmp) {
+        Pair<List<String>, List<String>> langPair = prepareDetectLanguage(cleanText, noopLangsTmp, preferredLangsTmp);
+        if (langPair == null) {
+            return new DetectedLanguage(null, new NoopLanguage());
+        }
+        List<String> additionalLangs = langPair.getLeft();
+        List<String> preferredLangs = langPair.getRight();
+
+        String[] words = cleanText.split(" ");
+        List<String> dominantLangCodes = UNICODE_BASED_LANG_IDENTIFIER.getDominantLangCodes(cleanText);
+        Map<String, Double> scores = new HashMap<>();
+        String detectionSource = "spellchecker";
+        for (Map.Entry<String, SpellingCheckRule> scp : spellingCheckRules.entrySet()) {
+            if (dominantLangCodes.contains(scp.getKey()) ^ (dominantLangCodes.isEmpty() && !NON_LATIN_CHARS_LANGUAGES.contains(scp.getKey()))) {
```

Comment:
`^` ?

Reply:
Javas XOR operator → Only check the language with this spellchecker if:
a) the spellchecker is from a language with non-Latin chars, AND the text contain non-Latin chars → **is true**
OR
b) The spellchecker is not from a language with non-Latin chars, AND the text does not contain any non-Latin chars.→ **is true**

Does not match if both are true, as a normal `or` would do. Could also be replaced by `!=`

Known issue:
Leads to problems with texts that contain non-Latin chars, but actually have a latin language as the main language. These would be misidentified.